File: examples/LayaAir/src/proto/ReqSceneEnter.ts
namespace proto {
export class ReqSceneEnter
{
	private _door_id: number;


	public Encode(): game.util.Packet {
		let packet: game.util.Packet = new game.util.Packet();
		packet.WriteUint(this._door_id);
		packet.Encode(2020);
		return packet;
	}


	public get door_id(): number { return this._door_id; }
	public set door_id(value: number) { this._door_id = value; }
}
}

File: examples/golang/server/proto/req_scene_enter.go
package proto

import (
	"packet"
)

type ReqSceneEnter struct {
	DoorId                   uint32
}

func ReqSceneEnterDecode(pack *packet.Packet) *ReqSceneEnter {
	reqSceneEnter := &ReqSceneEnter{}

	reqSceneEnter.DoorId = pack.ReadUint32()
	return reqSceneEnter
}
